### PR TITLE
Zscaler integration

### PR DESCRIPTION
**Pull request by Itay4**
Fixes https://github.com/demisto/etc/issues/10226

### CODE REVIEW

File: Integrations/integration-Zscaler.yml [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        blacklistUrls = url.split(',')
+        data = {
+            'blacklistUrls': blacklistUrls
+        }
+        res = http_request('POST', cmd_url, data, DEFAULT_HEADERS)
+        return 'Added to blacklist successfully.'
+
+    ''' EXECUTION CODE '''
+
+    auth = login()
```
Comment (anara123) [inline Integrations/integration-Zscaler.yml]: you don't use `auth` anywhere

File: Integrations/integration-Zscaler.yml [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+    LOG('command is %s' % (demisto.command(), ))
+    try:
+        if demisto.command() == 'test-module':
+            login()
+            demisto.results('ok')
+        elif demisto.command() == 'zcaler-blacklist-url':
+            demisto.results(blacklist_url(args.url))
```
Comment (anara123) [inline Integrations/integration-Zscaler.yml]: You defined `def blacklist_list` and here you use `blacklist_url`. Use `blacklist_url`

File: Integrations/integration-Zscaler.yml [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        return 'Added to blacklist successfully.'
+
+    ''' EXECUTION CODE '''
+
+    auth = login()
+
+    LOG('command is %s' % (demisto.command(), ))
+    try:
+        if demisto.command() == 'test-module':
+            login()
```
Comment (anara123) [inline Integrations/integration-Zscaler.yml]: if you use login before the `try` then you don't need this login

File: Integrations/integration-Zscaler.yml [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+    def login():
+        cmd_url = '/authenticatedSession'
+        data = {
+            'username': USERNAME,
+            'password': PASSWORD,
+            'apiKey': API_KEY,
+            'timestamp': int(round(time.time() * 1000))
+        }
+        res = http_request('POST', cmd_url, data, DEFAULT_HEADERS)
```
Comment (yaakovi) [inline Integrations/integration-Zscaler.yml]: you don't do anything with "res", so why keep it?

File: Integrations/integration-Zscaler.yml [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+    def logout():
+        cmd_url = '/authenticatedSession'
+        res = http_request('DELETE', cmd_url, None, DEFAULT_HEADERS)
```
Comment (yaakovi) [inline Integrations/integration-Zscaler.yml]: same goes here

File: Integrations/integration-Zscaler.yml [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+        logout()
+  type: python
+  commands:
+  - name: zcaler-blacklist-url
+    arguments:
+    - name: url
+      required: true
+      default: true
+      description: 'URLs to be blacklisted. Should be comma separated (i.e. snapchat.com,tinder.com)'
+    description: Adds a URL to the blacklist.
```
Comment (yaakovi) [inline Integrations/integration-Zscaler.yml]: be consistent with single or plural of arguments.
command description specify there is one url and the argument description specify a group

File: Integrations/integration-Zscaler.yml [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+        res = http_request('DELETE', cmd_url, None, DEFAULT_HEADERS)
+
+    def blacklist_url(url):
+        cmd_url = '/security/advanced/blacklistUrls?action=ADD_TO_LIST'
+        blacklistUrls = url.split(',')
+        data = {
+            'blacklistUrls': blacklistUrls
+        }
+        res = http_request('POST', cmd_url, data, DEFAULT_HEADERS)
+        return 'Added to blacklist successfully.'
```
Comment (yaakovi) [inline Integrations/integration-Zscaler.yml]: might be good to MD the list to the war room (easier to read than the command itself)

**Comment by anara123** [comment]
Merge when we have test instance

**Comment by anara123** [comment]
@Itay4 please test it

**Pull request merged**